packages/components/src/components/primitives/Breadcrumb/Breadcrumb.tsx

```diff
@@ -1,36 +1,69 @@
-import { useStyleConfig } from '@chakra-ui/react';
-import { Flex } from '@components';
-import React, { FC, Fragment } from 'react';
-import { IconChevronRight, Link, Text } from '..';
-import { BreadcrumbProps, BreadcrumbStyle } from './types';
+import React, { FC } from 'react';
+import { chakra, forwardRef, StyleProps, StylesProvider, useMultiStyleConfig, useStyles } from '@chakra-ui/react';
+import { getValidChildren } from '@chakra-ui/react-utils';
+import { Flex, IconChevronRight, Link, Text, FlexProps } from '@components';
+import { BreadcrumbProps, BreadcrumbStaticMembers } from './types';
 
-export const Breadcrumb: FC<BreadcrumbProps> = ({ items, separator, ...props }) => {
-  const copy = [...items];
-  const last = copy.pop();
+const LastItem = forwardRef((props, ref) => {
+  const styles = useStyles() as {
+    lastItem: StyleProps;
+  };
+  return <chakra.span aria-current="page" ref={ref} {...props} __css={styles.lastItem} />;
+});
 
-  const { container, separatorIcon, separator: separatorStyle, lastItem } = useStyleConfig(
-    'Breadcrumb'
-  ) as BreadcrumbStyle;
+const Separator = ({ icon, ...rest }) => {
+  const styles = useStyles() as {
+    separator: StyleProps;
+  };
+
+  let separator = <IconChevronRight {...rest} />;
+
+  if (icon) {
+    if (typeof icon === 'string') {
+      separator = <Text>{icon}</Text>;
+    } else {
+      separator = icon;
+    }
+  }
+
+  return React.cloneElement(separator, {
+    ...styles.separator,
+  });
+};
+
+const BreadcrumbItem = ({ isLastChild, separator, href, ...rest }) => {
+  if (isLastChild) {
+    return <LastItem {...rest} />;
+  }
+
+  return (
+    <>
+      <Link size="m" href={href} {...rest} />
+      <Separator icon={separator} />
+    </>
+  );
+};
+
+export const Breadcrumb: FC<BreadcrumbProps> & BreadcrumbStaticMembers = ({ children, separator, ...rest }) => {
+  const styles = useMultiStyleConfig('Breadcrumb', {}) as {
+    container: FlexProps;
```

Comment:
I wrote these types directly here since when I wanted to reuse a custom interface, an error related to incompatibility was thrown.